Add SolidJS Support (#934)
garden-co/jazz: docs/content/docs/reading/queries.mdx

| React/Expo | `useAll(query)`                | Returns `T[] \| undefined`  |
| Vue        | `useAll(query)`                | Returns a `Ref`             |
| Svelte     | `new QuerySubscription(query)` | Exposes reactive `.current` |

> | Solid      | `useAll(() => ({ query }))`    | Returns `T[] \| undefined`  |

Reviewer: ```suggestion
| Solid      | `useAll(() => ({ query }))`    | Returns `{ data, loading, error }` |
```

Looks like the actual return type is `{ data: T[] | undefined, loading: boolean, error: Error | unknown }`

### Fine-grained updates

Vue's `useAll`, Svelte's `QuerySubscription`, and Solid's `useAll` reconcile new query results into the existing reactive array in place rather than swapping the reference. When an upstream change touches a single row, only that row's affected fields write into the reactive proxy, so `$effect` (Svelte), Solid computations/effects, and `watch` / template bindings (Vue) only re-fire for components that depend on the fields that actually changed.

In practice this means:

- Adding, removing, or reordering rows updates the array structure only — untouched row objects keep their identity, so `{#each items as item (item.id)}` and `<TransitionGroup>` keyed renders are stable.
- Editing a single field on one row writes only that field — sibling rows do not re-render, and per-row components that read other fields stay quiet.
- Vue's `useAll` returns a deep `Ref` (not a `shallowRef`), so per-field reactivity composes with the rest of your component tree without manual unwrapping.

> - Solid's `useAll` returns an accessor function, so Solid's fine-grained dependency tracking only re-runs computations that read changed fields.

Reviewer: ```suggestion
- Solid's `useAll` returns a store, so Solid's fine-grained dependency tracking only re-runs computations that read changed fields.
```
